### PR TITLE
Fixed MongoDB TotalCount

### LINKED ISSUE #3683: Incorrect total count in MongoDbOffsetPaging
I'm using `MongoDbOffsetPaging` (with `MongoDbFiltering`) in my search query. I need total count of elements for generating pagination component so I added this option via `SetPagingOptions`. `totalCount` field appeared in query result but it's value is incorrect and depends on values of `take` and `skip` fields. 
Fore example, filters should return 30 items:
```
"take": 12,
"skip": 0 //items for first page

//returns: "totalCount": 13
```
```
"take": 12,
"skip": 12 //items for second page

//returns: "totalCount": 13
```
```
"take": 12,
"skip": 24 //items for third page

//returns: "totalCount": 6
```
In all examples I receive correct number of items (12, 12 and 6) but `totalCount` has different values (expected "30" in all cases). `MongoDbPaging` (cursor based pagination) returns correct `totalCount` but I can't use it for generation of pagination component.

### LINKED ISSUE #3868: totalCount seems to be wrong when using UseMongoDbPaging
**Describe the bug**
Hi everyone,

when i use UseMongoDbPaging with IncludeTotalCount=true , the totalCount field seems to be wrong when 'first' field is included in query.

**To Reproduce**
On MongoDB example (https://github.com/ChilliCream/hotchocolate-examples/tree/master/misc/MongoDB) ... When I write following query
```
{
  persons(where: { name: { contains: "Romeo" } }) {
    totalCount
    nodes {
      name
    }
  }
}
```
throws right number of items and totalCount (3):
```
{
  "data": {
    "persons": {
      "totalCount": 3,
      "nodes": [
        {
          "name": "Romeo McCloy"
        },
        {
          "name": "Romeo2 McCloy"
        },
        {
          "name": "Romeo3 McCloy"
        }
      ]
    }
  }
}
```
But when i include 'first' value, totalCount differs:

```
{
  persons(first:1, where: { name: { contains: "Romeo" } }) {
    totalCount
    nodes {
      name
    }
  }
}
```
```
{
  "data": {
    "persons": {
      "totalCount": 2,
      "nodes": [
        {
          "name": "Romeo McCloy"
        }
      ]
    }
  }
}

```

**Expected behavior**
totalCount should show always the total number of elements that matches the input filter, in above case, 3.

**Additional context**
Faced with HC 11.3 and also HC 12.0.0-preview.18

### PR DESCRIPTION
**Pull request by PascalSenn**
Closes #3868
Closes #3683

### CODE REVIEW

File: src/HotChocolate/MongoDb/src/Data/Paging/AggregateFluentPagingContainer.cs [hunk 1]
```diff
@@ -9,16 +9,17 @@ namespace HotChocolate.Data.MongoDb.Paging
     internal class AggregateFluentPagingContainer<TEntity> : IMongoPagingContainer<TEntity>
```
Comment (michaelstaib) [inline src/HotChocolate/MongoDb/src/Data/Paging/AggregateFluentPagingContainer.cs:9]: Line length?

File: src/HotChocolate/MongoDb/test/Data.MongoDb.Paging.Tests/MongoDbCursorPagingAggregateFluentTests.cs [hunk 2]
```diff
@@ -181,6 +181,25 @@ public async Task JustTotalCount()
...
         }
 
+        [Fact]
+        public async Task TotalCount_AndFirst()
+        {
+            Snapshot.FullName();
+
+            IRequestExecutor executor = await CreateSchemaAsync();
+
+
```
Comment (michaelstaib) [inline src/HotChocolate/MongoDb/test/Data.MongoDb.Paging.Tests/MongoDbCursorPagingAggregateFluentTests.cs:191]: line

File: src/HotChocolate/MongoDb/test/Data.MongoDb.Paging.Tests/MongoDbCursorPagingAggregateFluentTests.cs [hunk 3]
```diff
@@ -181,6 +181,25 @@ public async Task JustTotalCount()
...
+            IExecutionResult result = await executor
+                .ExecuteAsync(
+                    @"
```
Comment (michaelstaib) [inline src/HotChocolate/MongoDb/test/Data.MongoDb.Paging.Tests/MongoDbCursorPagingAggregateFluentTests.cs:194]: Formatting

File: src/HotChocolate/MongoDb/test/Data.MongoDb.Paging.Tests/MongoDbCursorPagingFindFluentTests.cs [hunk 4]
```diff
@@ -213,6 +213,25 @@ public async Task JustTotalCount()
             result.MatchDocumentSnapshot();
         }
 
+        [Fact]
+        public async Task TotalCount_AndFirst()
+        {
+            Snapshot.FullName();
+
+            IRequestExecutor executor = await CreateSchemaAsync();
+
```
Comment (michaelstaib) [inline src/HotChocolate/MongoDb/test/Data.MongoDb.Paging.Tests/MongoDbCursorPagingFindFluentTests.cs:222]: line

File: src/HotChocolate/MongoDb/test/Data.MongoDb.Paging.Tests/MongoDbCursorPagingFindFluentTests.cs [hunk 5]
```diff
@@ -213,6 +213,25 @@ public async Task JustTotalCount()
...
+
+            IExecutionResult result = await executor
+                .ExecuteAsync(
+                    @"
```
Comment (michaelstaib) [inline src/HotChocolate/MongoDb/test/Data.MongoDb.Paging.Tests/MongoDbCursorPagingFindFluentTests.cs:226]: formatting

File: src/HotChocolate/MongoDb/src/Data/Paging/AggregateFluentPagingContainer.cs (hunk 1 above)
Comment (PascalSenn) [reply src/HotChocolate/MongoDb/src/Data/Paging/AggregateFluentPagingContainer.cs:9]: 91

**Review by michaelstaib — APPROVED** [review]

**Pull request merged**